### ISSUE TITLE
bug in (quotient, Module, Ideal)

### ISSUE THREAD

**Issue by eisenbud** [Macaulay2/M2#2373]
The following code produces an error because the ambient free module of M
has a Schreyer order, whereas the ambient free module of 0*M does not (the free
modules mentioned in the error message are the same except for this.)

```restart
S = ZZ/101[x]
I = ideal "x2"
R=S/I
F = res coker vars R
M = image F.dd_2
(0*M):(ideal gens R)
stdio:6:6:(3): error: expected matrices with the same target
```

**Comment by mahrud** [comment]
Yes, @jchen419 and I traced this to using `=!=` instead of `!=` for checking equality of modules in `remainder(Matrix,Matrix)`. I made an issue in https://github.com/Macaulay2/M2/issues/1726 with a list of all uses of `=!=` but didn't hear back about whether there's a reason `=!=` has been used in so many places instead of `!=`.

**Comment by DanGrayson** [comment]
It's probably fine to change those occurrences of `=!=`.  Perhaps simultaneously review the method for `Module == Module` for speed -- if the two modules have identical generators and relations, it seems that it could return `true` more quickly.
